test/support/test_notifier.ex

```diff
@@ -1,5 +1,10 @@
 defmodule CoursePlanner.TestNotifier do
+
+  alias CoursePlanner.{Notification, Notifier.Server}
+
   def notify_user(_), do: :ok
-  def notify_later(_), do: :ok
+  def notify_later(%Notification{} = notification) do
+    Server.handle_cast({:save_email, notification}, "test")
+  end
```

Comment:
This `TestNotifier` module is a fake module to supress errors during test, so you're not supposed to call the "real" notifier here.

Reply:
then who am i suppose to test whether the record is created or not ?

Reply:
I think i disagree, the notification should be tested black boxed. i cannot during my test rely on the fact that it works

Reply:
I think there was a discussion before when I raised this issue, @ggpasqualino suggested to test the notifier separately (which is already being done). You separate the part that test the workflow of creating the notification and the part that persists.

Reply:
but I am not testing the emails, I am only testing the function i wrote.
i agree with testing notifications separately cause they are written separately, but for my case i am testing "notify_missing_attendances" function

Reply:
Yes, in this case you should test the "query and build the missing attendances" and "persist the created notifications" separately.
But if the way you wrote it is not raising errors, I guess it's fine. The fake module was being used because, when testing controller calls, it was raising errors because the repo process was not owned by the calling process.

Reply:
I think the main issue is when we actually notify, the gen sever talks to swoosh